compute-coverage.sh

```diff
@@ -17,8 +17,22 @@ set -Eeuxo pipefail
 SUM_TIX="all.tix"
 hpc sum --output=$SUM_TIX --union --exclude=Main --exclude=GitRev $(find . -name "*.tix")
 
-# Generate report
-HPC_ROOT=$(find dist-newstyle -name "hpc")
+# Find the HPC dir, and don't trip on old versions after a version bump.
+# See saw-script #2114.
+#
+# There is no direct way to do this. Instead, fetch the name of the
+# SAW executable. This gives us a path into the build directory for
+# the current version:
+#    dist-newstyle/build/$TARGET/ghc-$GHC/saw-script-$VERSION/build/saw/saw
+#
+# where we don't want to have to try to figure out $TARGET, $GHC, or $VERSION.
+#
+# The hpc dir we want is parallel to the build dir, so drop the end bits off
+# and replace them.
+SAW=$(cabal list-bin -v0 exe:saw)
+HPC_ROOT=$(echo "$SAW" | sed 's,build/saw/saw$,hpc,')
```

Comment:
Out of curiosity, I asked the folks on the #ghc Matrix channel if this particular subdirectory was liable to change in future `cabal` versions. I'm glad I asked, because the location actually [changed in `cabal-install-3.12.0.0`](https://github.com/haskell/cabal/pull/9464). In that version, `cabal` now puts generated `hpc`-related files under a subdirectory that looks something like:

```
/home/ryanscott/Documents/Hacking/Haskell/saw-script/dist-newstyle/build/x86_64-linux/ghc-9.4.8/saw-script-1.0.0.99/build/extra-compilation-artifacts/hpc/vanilla/mix/saw-script-1.0.0.99-inplace
```

That is to say, they're now located under `saw-script-<ver>/build` rather than `saw-script-<ver>/hpc`. It might be worth calling `find` on `saw-script-<ver>` instead to be forward-compatible with `cabal-install-3.12.0.0` or later.

Reply:
Hmm, yes indeed.

Reply:
Ok, it now does that.